tools/s3-gc-stats

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+
+# Assumptions made by this code:
+# - list_object_versions sorts all versions (including delete markers) by key
+#   (in ascending order) and last modified date (in descending order, sometimes
+#   with consecutive equal timestamps) and returns them in chunks of 1000 (by
+#   default), with each chunk divided into proper versions and delete markers.
```

Comment:
note: these assumptions seems to be validated in the code below: https://github.com/dandi/dandi-cli/pull/473/files#diff-c840778a73d440599aee9be998e0b555e84831c627e589f54439e8d3f2819295R103 . Great!